web/src/main/java/org/mskcc/cbio/portal/web/ApiController.java

```diff
@@ -146,14 +146,19 @@
     
     @Transactional
     @RequestMapping(value = "/samples", method = {RequestMethod.GET, RequestMethod.POST})
-    public @ResponseBody List<DBSample> getSamples(@RequestParam(required = true) String study_id, @RequestParam(required = false) List<String> sample_ids) {
+    public @ResponseBody List<DBSample> getSamples(@RequestParam(required = true) String study_id, @RequestParam(required = false) List<String> sample_ids, @RequestParam(required = false) List<String> patient_ids, @RequestParam(required = false) String patient_id) {
 	    if (sample_ids != null) {
 		    return service.getSamples(study_id, sample_ids);
-	    } else {
+	    } else if (patient_ids != null) {
+                    return service.getSamplesByStudyAndPatientId(study_id,patient_ids);
+            } else if (patient_id != null) {
+                    return service.getSamplesByStudyAndPatientId(study_id,patient_id);
```

Comment:
Do we really need this? A GET request will look exactly the same (?patient_ids=id1  vs ?patient_id=id1) and for POST you'd only need to wrap it in brackets, not a huge deal. Is there another reason to keep this?